### PR TITLE
optimized test tweaks

### PR DESCRIPTION
**Pull request by czue**
@millerdev

### CODE REVIEW

File: corehq/tests/optimizer.py [hunk 1]
```diff
@@ -115,28 +116,25 @@ def _extract_tests(test_suite_or_test):
...
+    _real_installed_apps = settings.INSTALLED_APPS
+    needed_apps = test_map.get_needed_installed_apps()
+    print 'overriding settings.INSTALLED_APPS to {}'.format(
+        ','.join(test_map.get_needed_installed_apps())
+    )
+    settings.INSTALLED_APPS = tuple(needed_apps)
+    apps.set_installed_apps(settings.INSTALLED_APPS)
+    yield
+    settings.INSTALLED_APPS = _real_installed_apps
+    apps.unset_installed_apps()
```
Comment (millerdev) [inline corehq/tests/optimizer.py]: One tricky thing about using `@contextmanager` is ensuring that finalization occurs if an exception is raised. This likely should be changed to
```py
...
try:
    yield
finally:
    settings.INSTALLED_APPS = _real_installed_apps
    apps.unset_installed_apps()
```

Comment (czue) [inline corehq/tests/optimizer.py]: makes sense - changed.

**Comment by millerdev** [comment]
:+1: on test pass

**Comment by czue** [comment]
self-merging

**Pull request merged**